python/sbp/sbp2json.py

```diff
@@ -69,56 +60,12 @@ def dump(args, res):
     sys.stdout.write("\n")
 
 
-def sbp_main(args):
-    header_len = 6
-    reader = io.open(args.file.fileno(), 'rb')
-    buf = memoryview(bytearray(4096))
-    unconsumed_offset = 0
-    read_offset = 0
-    buffer_remaining = len(buf)
-    include = set(args.include)
-    while True:
-        if buffer_remaining == 0:
-            buf[0:(read_offset - unconsumed_offset)] = buf[unconsumed_offset:read_offset]
-            read_offset = read_offset - unconsumed_offset
-            unconsumed_offset = 0
-            buffer_remaining = len(buf) - read_offset
-        mv = buf[read_offset:]
-        read_length = reader.readinto(mv)
-        if read_length == 0:
-            unconsumed = read_offset - unconsumed_offset
-            if unconsumed != 0:
-                sys.stderr.write("unconsumed: {}\n".format(unconsumed))
-                sys.stderr.flush()
-            break
-        read_offset += read_length
-        buffer_remaining -= read_length
-        while True:
-            bytes_available = read_offset - unconsumed_offset
-            b = buf[unconsumed_offset:(unconsumed_offset + bytes_available)]
-            if len(b) == 0:
-                break
-            if b[0] != SBP_PREAMBLE:
-                consumed = 1
-            else:
-                try:
-                    m = sbp.msg.SBP.unpack(b)
-                    if not include or m.msg_type in include:
-                        m = sbp.table.dispatch(m)
-                        dump(args, m)
-                    consumed = header_len + m.length + 2
-                except (UnpackError, StreamError):
-                    break
-                except ValueError:
-                    consumed = 1
-            unconsumed_offset += consumed
-
-
 def module_main():
-    args = get_args()
-    if not args:
-        sys.exit(1)
-    try:
-        sbp_main(args)
-    except KeyboardInterrupt:
-        pass
+    if args := get_args():
+        with Handler(Framer(args.file.read, None, verbose=True)) as source:
+            try:
+                it = source.__iter__()
```

Comment:
It's considered unpythonic (whatever that means) to manually call "dunder" methods.  Use `iter(source)` over `source.__iter__()`.  Better yet use a for loop here:

```python
for msg_meta in source:
    dump(msg_meta)
```

or 

```python
for msg_meta in source:
    if msg_meta is None:
        break
    dump(msg_meta)
```

instead depending on what behaviour you want.

Also don't use the walrus operator unless you are sure that this file will only run on Python >= 3.9!

Reply:
i tried doing that before but the order kept changing when i ran it, but now when i run it, it changes regardless 

also didnt know it was called the Walrus Operator